src/background-plugin.ts

```diff
@@ -272,27 +291,122 @@ class BackgroundPlugin {
       const [, reject] = this.#request as any[];
       this.#request = null;
       return reject(err);
+    } finally {
+      clearInterval(timer);
     }
   }
+}
 
-  async getExports(name?: string): Promise<WebAssembly.ModuleExportDescriptor[]> {
-    return await this.#invoke('getExports', name ?? '0');
+// Return control to the waiting promise. Anecdotally, this appears to help
+// with a race condition in Bun.
+const MAX_WAIT = 500;
+class RingBufferWriter {
+  output: SharedArrayBuffer;
+  scratch: ArrayBuffer;
+  scratchView: DataView;
+  outputOffset: number;
+  flag: Int32Array;
+
+  static SAB_IDX = 0;
+
+  constructor(output: SharedArrayBuffer) {
+    this.scratch = new ArrayBuffer(8);
+    this.scratchView = new DataView(this.scratch);
+    this.output = output;
+    this.outputOffset = SAB_BASE_OFFSET;
+    this.flag = new Int32Array(this.output);
+    this.wait(0);
```

Comment:
This is the "first bug" I talked about – now we `.wait(0)` before attempting to write anything.